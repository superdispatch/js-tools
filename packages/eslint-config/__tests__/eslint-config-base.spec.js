const { testInheritance } = require('../__testutils__/test-eslint-config');

const configFile = require.resolve('../base');

testInheritance(configFile);
