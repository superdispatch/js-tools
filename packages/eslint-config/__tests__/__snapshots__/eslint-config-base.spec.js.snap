// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inheritance config 1`] = `
Object {
  "env": Object {},
  "globals": Object {},
  "parser": null,
  "parserOptions": Object {
    "ecmaVersion": 2018,
  },
  "plugins": Array [
    "prettier",
    "import",
  ],
  "settings": Object {},
}
`;

exports[`Inheritance dev rules 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ --- --- @@
    "arrow-body-style": Array [
-     1,
+     2,
      "as-needed",
@@ --- --- @@
    "curly": Array [
-     1,
+     2,
      "all",
@@ --- --- @@
    "dot-notation": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "no-confusing-arrow": Array [
-     1,
+     2,
      Object {
@@ --- --- @@
    "no-div-regex": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "no-else-return": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "no-floating-decimal": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "no-implicit-coercion": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "no-lonely-if": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "no-undef-init": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "no-unneeded-ternary": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "no-useless-computed-key": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "no-useless-rename": Array [
-     1,
+     2,
    ],
    "no-useless-return": Array [
-     1,
+     2,
    ],
    "no-var": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "object-shorthand": Array [
-     1,
+     2,
      "always",
@@ --- --- @@
    "operator-assignment": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "prefer-arrow-callback": Array [
-     1,
+     2,
      Object {
@@ --- --- @@
    "prefer-destructuring": Array [
-     1,
+     2,
      Object {
@@ --- --- @@
    "prefer-numeric-literals": Array [
-     1,
+     2,
    ],
    "prefer-object-spread": Array [
-     1,
+     2,
    ],
    "prefer-template": Array [
-     1,
+     2,
    ],
    "prettier/prettier": Array [
-     1,
+     2,
    ],
@@ --- --- @@
    "yoda": Array [
-     1,
+     2,
      "never",
`;

exports[`Inheritance rules 1`] = `
Object {
  "array-bracket-newline": Array [
    "off",
  ],
  "array-bracket-spacing": Array [
    "off",
  ],
  "array-element-newline": Array [
    "off",
  ],
  "arrow-body-style": Array [
    1,
    "as-needed",
    Object {
      "requireReturnForObjectLiteral": false,
    },
  ],
  "arrow-parens": Array [
    "off",
  ],
  "arrow-spacing": Array [
    "off",
  ],
  "block-spacing": Array [
    "off",
  ],
  "brace-style": Array [
    "off",
  ],
  "camelcase": Array [
    2,
    Object {
      "ignoreDestructuring": true,
      "properties": "never",
    },
  ],
  "comma-dangle": Array [
    "off",
  ],
  "comma-spacing": Array [
    "off",
  ],
  "comma-style": Array [
    "off",
  ],
  "computed-property-spacing": Array [
    "off",
  ],
  "constructor-super": Array [
    "error",
  ],
  "curly": Array [
    1,
    "all",
  ],
  "dot-location": Array [
    "off",
  ],
  "dot-notation": Array [
    1,
  ],
  "eol-last": Array [
    "off",
  ],
  "eqeqeq": Array [
    1,
    "smart",
  ],
  "for-direction": Array [
    "error",
  ],
  "func-call-spacing": Array [
    "off",
  ],
  "func-names": Array [
    1,
    "as-needed",
  ],
  "function-paren-newline": Array [
    "off",
  ],
  "generator-star": Array [
    "off",
  ],
  "generator-star-spacing": Array [
    "off",
  ],
  "getter-return": Array [
    "error",
  ],
  "implicit-arrow-linebreak": Array [
    "off",
  ],
  "import/default": Array [
    2,
  ],
  "import/export": Array [
    2,
  ],
  "import/first": Array [
    1,
  ],
  "import/named": Array [
    2,
  ],
  "import/namespace": Array [
    2,
  ],
  "import/no-anonymous-default-export": Array [
    1,
    Object {
      "allowAnonymousClass": false,
      "allowAnonymousFunction": false,
      "allowArray": false,
      "allowArrowFunction": false,
      "allowCallExpression": false,
      "allowLiteral": false,
      "allowObject": false,
    },
  ],
  "import/no-deprecated": Array [
    2,
  ],
  "import/no-duplicates": Array [
    1,
  ],
  "import/no-mutable-exports": Array [
    1,
  ],
  "import/no-named-as-default": Array [
    1,
  ],
  "import/no-named-as-default-member": Array [
    1,
  ],
  "import/no-unresolved": Array [
    2,
  ],
  "indent": Array [
    "off",
  ],
  "indent-legacy": Array [
    "off",
  ],
  "jsx-quotes": Array [
    "off",
  ],
  "key-spacing": Array [
    "off",
  ],
  "keyword-spacing": Array [
    "off",
  ],
  "linebreak-style": Array [
    "off",
  ],
  "lines-around-comment": Array [
    0,
  ],
  "max-len": Array [
    0,
  ],
  "multiline-ternary": Array [
    "off",
  ],
  "new-parens": Array [
    "off",
  ],
  "newline-per-chained-call": Array [
    "off",
  ],
  "no-alert": Array [
    1,
  ],
  "no-arrow-condition": Array [
    "off",
  ],
  "no-async-promise-executor": Array [
    "error",
  ],
  "no-case-declarations": Array [
    "error",
  ],
  "no-class-assign": Array [
    "error",
  ],
  "no-comma-dangle": Array [
    "off",
  ],
  "no-compare-neg-zero": Array [
    "error",
  ],
  "no-cond-assign": Array [
    0,
  ],
  "no-confusing-arrow": Array [
    1,
    Object {
      "allowParens": true,
    },
  ],
  "no-console": Array [
    1,
  ],
  "no-const-assign": Array [
    "error",
  ],
  "no-constant-condition": Array [
    "error",
  ],
  "no-control-regex": Array [
    "error",
  ],
  "no-debugger": Array [
    "error",
  ],
  "no-delete-var": Array [
    "error",
  ],
  "no-div-regex": Array [
    1,
  ],
  "no-dupe-args": Array [
    "error",
  ],
  "no-dupe-class-members": Array [
    "error",
  ],
  "no-dupe-keys": Array [
    "error",
  ],
  "no-duplicate-case": Array [
    "error",
  ],
  "no-else-return": Array [
    1,
  ],
  "no-empty": Array [
    1,
    Object {
      "allowEmptyCatch": true,
    },
  ],
  "no-empty-character-class": Array [
    "error",
  ],
  "no-empty-pattern": Array [
    "error",
  ],
  "no-ex-assign": Array [
    "error",
  ],
  "no-extra-boolean-cast": Array [
    "error",
  ],
  "no-extra-parens": Array [
    "off",
  ],
  "no-extra-semi": Array [
    "off",
  ],
  "no-fallthrough": Array [
    "error",
  ],
  "no-floating-decimal": Array [
    1,
  ],
  "no-func-assign": Array [
    "error",
  ],
  "no-global-assign": Array [
    "error",
  ],
  "no-implicit-coercion": Array [
    1,
  ],
  "no-inner-declarations": Array [
    "error",
  ],
  "no-invalid-regexp": Array [
    "error",
  ],
  "no-irregular-whitespace": Array [
    "error",
  ],
  "no-lonely-if": Array [
    1,
  ],
  "no-misleading-character-class": Array [
    "error",
  ],
  "no-mixed-operators": Array [
    0,
  ],
  "no-mixed-spaces-and-tabs": Array [
    "off",
  ],
  "no-multi-spaces": Array [
    "off",
  ],
  "no-multiple-empty-lines": Array [
    "off",
  ],
  "no-new-symbol": Array [
    "error",
  ],
  "no-obj-calls": Array [
    "error",
  ],
  "no-octal": Array [
    "error",
  ],
  "no-prototype-builtins": Array [
    "error",
  ],
  "no-redeclare": Array [
    "error",
  ],
  "no-regex-spaces": Array [
    "error",
  ],
  "no-reserved-keys": Array [
    "off",
  ],
  "no-self-assign": Array [
    "error",
  ],
  "no-shadow": Array [
    2,
    Object {
      "builtinGlobals": true,
      "hoist": "functions",
    },
  ],
  "no-shadow-restricted-names": Array [
    "error",
  ],
  "no-space-before-semi": Array [
    "off",
  ],
  "no-spaced-func": Array [
    "off",
  ],
  "no-sparse-arrays": Array [
    "error",
  ],
  "no-tabs": Array [
    0,
  ],
  "no-this-before-super": Array [
    "error",
  ],
  "no-trailing-spaces": Array [
    "off",
  ],
  "no-undef": Array [
    "error",
  ],
  "no-undef-init": Array [
    1,
  ],
  "no-unexpected-multiline": Array [
    0,
  ],
  "no-unneeded-ternary": Array [
    1,
  ],
  "no-unreachable": Array [
    "error",
  ],
  "no-unsafe-finally": Array [
    "error",
  ],
  "no-unsafe-negation": Array [
    "error",
  ],
  "no-unused-labels": Array [
    "error",
  ],
  "no-unused-vars": Array [
    "error",
  ],
  "no-use-before-define": Array [
    1,
    Object {
      "functions": false,
    },
  ],
  "no-useless-catch": Array [
    "error",
  ],
  "no-useless-computed-key": Array [
    1,
  ],
  "no-useless-escape": Array [
    "error",
  ],
  "no-useless-rename": Array [
    1,
  ],
  "no-useless-return": Array [
    1,
  ],
  "no-var": Array [
    1,
  ],
  "no-warning-comments": Array [
    2,
    Object {
      "location": "anywhere",
      "terms": Array [
        "todo",
        "fixme",
        "@ts-ignore",
      ],
    },
  ],
  "no-whitespace-before-property": Array [
    "off",
  ],
  "no-with": Array [
    "error",
  ],
  "no-wrap-func": Array [
    "off",
  ],
  "nonblock-statement-body-position": Array [
    "off",
  ],
  "object-curly-newline": Array [
    "off",
  ],
  "object-curly-spacing": Array [
    "off",
  ],
  "object-property-newline": Array [
    "off",
  ],
  "object-shorthand": Array [
    1,
    "always",
    Object {
      "avoidQuotes": true,
    },
  ],
  "one-var-declaration-per-line": Array [
    "off",
  ],
  "operator-assignment": Array [
    1,
  ],
  "operator-linebreak": Array [
    "off",
  ],
  "padded-blocks": Array [
    "off",
  ],
  "prefer-arrow-callback": Array [
    1,
    Object {
      "allowNamedFunctions": false,
      "allowUnboundThis": true,
    },
  ],
  "prefer-destructuring": Array [
    1,
    Object {
      "AssignmentExpression": Object {
        "array": false,
        "object": false,
      },
      "VariableDeclarator": Object {
        "array": true,
        "object": true,
      },
    },
    Object {
      "enforceForRenamedProperties": false,
    },
  ],
  "prefer-numeric-literals": Array [
    1,
  ],
  "prefer-object-spread": Array [
    1,
  ],
  "prefer-template": Array [
    1,
  ],
  "prettier/prettier": Array [
    1,
  ],
  "quote-props": Array [
    "off",
  ],
  "quotes": Array [
    0,
  ],
  "require-atomic-updates": Array [
    "error",
  ],
  "require-yield": Array [
    "error",
  ],
  "rest-spread-spacing": Array [
    "off",
  ],
  "semi": Array [
    "off",
  ],
  "semi-spacing": Array [
    "off",
  ],
  "semi-style": Array [
    "off",
  ],
  "space-after-function-name": Array [
    "off",
  ],
  "space-after-keywords": Array [
    "off",
  ],
  "space-before-blocks": Array [
    "off",
  ],
  "space-before-function-paren": Array [
    "off",
  ],
  "space-before-function-parentheses": Array [
    "off",
  ],
  "space-before-keywords": Array [
    "off",
  ],
  "space-in-brackets": Array [
    "off",
  ],
  "space-in-parens": Array [
    "off",
  ],
  "space-infix-ops": Array [
    "off",
  ],
  "space-return-throw-case": Array [
    "off",
  ],
  "space-unary-ops": Array [
    "off",
  ],
  "space-unary-word-ops": Array [
    "off",
  ],
  "switch-colon-spacing": Array [
    "off",
  ],
  "template-curly-spacing": Array [
    "off",
  ],
  "template-tag-spacing": Array [
    "off",
  ],
  "unicode-bom": Array [
    "off",
  ],
  "use-isnan": Array [
    "error",
  ],
  "valid-typeof": Array [
    "error",
  ],
  "wrap-iife": Array [
    "off",
  ],
  "wrap-regex": Array [
    "off",
  ],
  "yield-star-spacing": Array [
    "off",
  ],
  "yoda": Array [
    1,
    "never",
    Object {
      "exceptRange": true,
      "onlyEquality": false,
    },
  ],
}
`;
